engine/crates/engine-v2/src/sources/graphql/federation.rs

```diff
@@ -151,6 +145,59 @@ impl FederationEntityResolver {
     }
 }
 
+async fn cache_fetches<'ctx, R: Runtime>(
+    ctx: ExecutionContext<'ctx, R>,
+    endpoint: schema::SchemaWalker<'_, GraphqlEndpointId>,
+    headers: &http::HeaderMap,
+    representations: Vec<Box<RawValue>>,
+    ingester: &mut EntityIngester<'ctx, R>,
+) -> CacheFetchOutcome {
+    let header_result = serde_json::to_vec(
```

Comment:
Nitpick: would have avoided the JSON by hashing directly the name & value (with headers length, name/value length or null separator for name).